### PR TITLE
JAMES-2671 Re-enable partial integration test execution for distributed James

### PR DESCRIPTION
**Pull request by chibenwa**

### CODE REVIEW

File: server/protocols/jmap-integration-testing/memory-jmap-integration-testing/pom.xml [hunk 1]
```diff
@@ -177,18 +177,4 @@
...
 
-    <build>
-        <plugins>
-            <plugin>
-                <groupId>org.apache.maven.plugins</groupId>
-                <artifactId>maven-surefire-plugin</artifactId>
-                <configuration>
-                    <reuseForks>false</reuseForks>
-                    <forkCount>1C</forkCount>
-                    <excludedGroups combine.self="override">org.apache.james.jmap.categories.CassandraAndElasticSearchOnly</excludedGroups>
```
Comment (rouazana) [inline server/protocols/jmap-integration-testing/memory-jmap-integration-testing/pom.xml]: I guess there are here for a good reason: CassandraAndElasticSearch only are for tests that does not work with a basic in memory implementation, like for example stemming.

Comment (chibenwa) [reply server/protocols/jmap-integration-testing/memory-jmap-integration-testing/pom.xml]: I think what you says don't make sense as CassandraAndElasticSearch is a subset of tests that we want to execute against cassandra+es and not a subset of test designed for Cassandra+ES

Won't fight on this - I just want my build time down to a reasonable amount.

Comment (rouazana) [reply server/protocols/jmap-integration-testing/memory-jmap-integration-testing/pom.xml]: why is it named CassandraAndElasticSearchOnly then?

Comment (rouazana) [reply server/protocols/jmap-integration-testing/memory-jmap-integration-testing/pom.xml]: oh strange, it's not used at all?

Comment (chibenwa) [reply server/protocols/jmap-integration-testing/memory-jmap-integration-testing/pom.xml]: For running the subset that we want to run cassandra +es on

**Comment by mbaechler** [comment]
So, we consider that the product on which we put the maximum effort doesn't need to be the more tested?

**Comment by trantienduchn** [comment]
> So, we consider that the product on which we put the maximum effort doesn't need to be the more tested?

As I remember, the job to merge PRs to master, it run all tests

**Comment by chibenwa** [comment]
> So, we consider that the product on which we put the maximum effort doesn't need to be the more tested?

Well playing these tests have a huge cost both on build time and stability cf recent CI results

This endanger overall developments, and we aggreed before on having "Smoke tests" running on non memory implementations.

On this branch:

```
[INFO] Apache James :: Server :: JMAP :: Cassandra Integration testing SUCCESS [04:13 min]
[INFO] Apache James :: Server :: JMAP :: Memory Integration testing SUCCESS [04:40 min]
[INFO] Apache James :: Server :: JMAP :: RabbitMQ + Object Store + Cassandra Integration testing SUCCESS [12:15 min]
[INFO] Apache James :: Server :: Web Admin server integration tests SUCCESS [03:07 min]
[INFO] Apache James :: Server :: Web Admin 3.4.0-SNAPSHOT . SUCCESS [  0.568 s]
[INFO] ------------------------------------------------------------------------
[INFO] BUILD SUCCESS
```

Without the smoke tests:

```
[INFO] Apache James :: Server :: JMAP :: Cassandra Integration testing SUCCESS [04:11 min]
[INFO] Apache James :: Server :: JMAP :: Memory Integration testing SUCCESS [04:29 min]
[INFO] Apache James :: Server :: JMAP :: RabbitMQ + Object Store + Cassandra Integration testing SUCCESS [  02:06 h]
[INFO] Apache James :: Server :: Web Admin server integration tests SUCCESS [03:01 min]
[INFO] Apache James :: Server :: Web Admin 3.4.0-SNAPSHOT . SUCCESS [  0.574 s]
[INFO] ------------------------------------------------------------------------
[INFO] BUILD SUCCESS
[INFO] ------------------------------------------------------------------------
[INFO] Total time: 04:39 h

```

If you find a solution to run these tests more quickly then sure let's re-enable it and reopen the topic to discussion. As far as I am aware of, nothing really changed on that topic. Then I don't see the need of discussing it once again.

**Comment by rouazana** [comment]
> So, we consider that the product on which we put the maximum effort doesn't need to be the more tested?

This product will be well tested. The tests that are not run are mostly protocol related tests which does not depend at all on the backend.

**Comment by chibenwa** [comment]
Merged

**Pull request closed without merging**